pkg/instancetype/controller/vm/controller.go

```diff
@@ -159,27 +159,35 @@ func (c *controller) handleExpand(
 	}
 
 	// Only update the VM if we have changed something by applying an instance type and preference
-	if !equality.Semantic.DeepEqual(vm.Spec, expandVMCopy.Spec) {
-		updatedVM, err := c.clientset.VirtualMachine(expandVMCopy.Namespace).Update(context.Background(), expandVMCopy, metav1.UpdateOptions{})
+	if !equality.Semantic.DeepEqual(vm, expandVMCopy) {
+		_, err := c.clientset.VirtualMachine(expandVMCopy.Namespace).Update(
+			context.Background(), expandVMCopy, metav1.UpdateOptions{})
```

Comment:
nit: it may be more stable to use the result of this call, because there may be a webhook that modifies it. Maybe like this?
```golang
updatedVM, err := c.clientset.VirtualMachine(expandVMCopy.Namespace).Update(
	context.Background(), expandVMCopy, metav1.UpdateOptions{})
if err != nil {
	return vm, fmt.Errorf("error encountered when trying to update expanded VirtualMachine: %v", err)
}

updatedVM.Status = expandVMCopy.Status
updatedVM, err = c.clientset.VirtualMachine(updatedVM.Namespace).UpdateStatus(
	context.Background(), updatedVM, metav1.UpdateOptions{})
if err != nil {
	return vm, fmt.Errorf("error encountered when trying to update expanded VirtualMachine Status: %v", err)
}
```